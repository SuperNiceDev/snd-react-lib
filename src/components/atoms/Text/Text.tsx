import React, { MouseEvent, ReactNode } from "react";

import css from "./Text.module.scss";

export enum Tags {
  div = "div",
  span = "span",
  label = "label",
  p = "p",
  h1 = "h1",
  h2 = "h2",
  h3 = "h3",
  h4 = "h4",
  h5 = "h5",
  h6 = "h6",
}

export type TextProps = {
  className?: string;
  id?: string;
  htmlFor?: string;
  Tag?: Tags;
  tagToReplace?: Tags;
  text?: string;
  style?: React.CSSProperties;
  onClick?: (evt: MouseEvent) => void;
  children?: ReactNode;
};

const Text = ({
  className,
  id,
  htmlFor,
  Tag = Tags.p,
  text,
  style,
  onClick = () => null,
  children,
}: TextProps) => {
  // console.log("text: ", text);

  return (
    <Tag
      className={`${css.root}${className ? ` ${className}` : ""}`}
      id={id}
      style={style}
      onClick={onClick}
      htmlFor={htmlFor}
    >
      {children || text}
      {/* XXXXXX */}
    </Tag>
  );
};

export default Text;
