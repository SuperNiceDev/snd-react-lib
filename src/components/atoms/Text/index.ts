export { default, Tags } from "./Text";
export type { IText } from "./Text";
