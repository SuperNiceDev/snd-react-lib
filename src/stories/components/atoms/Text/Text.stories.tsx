import React from "react";

// import Text, { Tags } from "@src/components/atoms/Text";
import Text, { Tags, TextProps } from "@src/components/atoms/Text";
import type { Meta, StoryObj } from "@storybook/react";

import css from "./Text.stories.module.scss";

// type TextProps = typeof Text;

export default {
  component: Text,
  argTypes: {
    Tag: {
      name: "Tag: ",
      description: "",
      defaultValue: Tags.p,
      control: "inline-radio",
      options: [
        Tags.div,
        Tags.span,
        Tags.label,
        Tags.p,
        Tags.h1,
        Tags.h2,
        Tags.h3,
        Tags.h4,
        Tags.h5,
        Tags.h6,
      ],
    },
  },
} as Meta<TextProps>;

type StoryProps = StoryObj<TextProps>;

const Template = (args: StoryProps) => {
  return (
    <div className={css.root}>
      <Text {...args} />
    </div>
  );
};

export const Story1 = {
  name: "Text",
  render: (args: StoryProps) => <Template {...args} />,
  args: {
    className: "myText",
    Tag: Tags.p,
    text: "my text",
  },
};
