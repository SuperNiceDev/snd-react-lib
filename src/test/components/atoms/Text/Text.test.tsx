import React from "react";

import { render, screen } from "@testing-library/react";
import Text from "snd-react-lib/components/atoms/Text";

it("matches the snapshot", () => {
  const { asFragment } = render(<Text>My text</Text>);
  expect(asFragment()).toMatchSnapshot();
});

test("contains correct placeholder", () => {
  render(<Text text="my text" />);
  const text = screen.getByText("my text");

  expect(text).toBeInTheDocument();
});
