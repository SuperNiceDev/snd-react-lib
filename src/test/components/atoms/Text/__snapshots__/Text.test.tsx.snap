// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <p
    class="root"
  >
    My text
  </p>
</DocumentFragment>
`;
